lib/hammer_cli_foreman/architecture.rb

```diff
@@ -11,6 +11,12 @@ class ListCommand < HammerCLIForeman::ListCommand
         field :name, _("Name")
       end
 
+      def option_sources
+        sources = super
+        context[:defaults]&.defaults_settings&.clear
```

Comment:
This can have unpleasant consequences:
 - What if default settings contain not only taxonomy related options? `.clear` will wipe all of it.
 - Try to set defaults then run `hammer shell` > `defaults list` > `architecture list` > `defaults list`. The defaults have been removed, so the next command which supports taxonomy related options will run without them.

Reply:
@ofedoren Thanks for the review. You are definitely right and I am working on fixing it.